File: lib/spiral/platform_.py
from collections.abc import Callable
from typing import Any
from fable_modules.fable_library.reflection import (TypeInfo, union_type)
from fable_modules.fable_library.types import (Array, Union)

def _expr125() -> TypeInfo:
    return union_type("Platform.US0", [], US0, lambda: [[], [], []])


class US0(Union):
    __slots__ = ["tag", "fields"]
    def __init__(self, tag: int, *fields: Any) -> None:
        super().__init__()
        self.tag: int = tag or 0
        self.fields: Array[Any] = list(fields)

    @staticmethod
    def cases() -> list[str]:
        return ["US0_0", "US0_1", "US0_2"]


US0_reflection = _expr125

def _expr126() -> TypeInfo:
    return union_type("Platform.US1", [], US1, lambda: [[("f0_0", US0_reflection())], [("f1_0", US0_reflection())], [("f2_0", US0_reflection())], [("f3_0", US0_reflection())], [("f4_0", US0_reflection())]])


class US1(Union):
    __slots__ = ["tag", "fields"]
    def __init__(self, tag: int, *fields: Any) -> None:
        super().__init__()
        self.tag: int = tag or 0
        self.fields: Array[Any] = list(fields)

    @staticmethod
    def cases() -> list[str]:
        return ["US1_0", "US1_1", "US1_2", "US1_3", "US1_4"]


US1_reflection = _expr126

def closure0(unit_var: None, unit_var_1: None) -> bool:
    raise Exception(("platform.is_windows / target: " + str(US1(4, US0(0)))) + "")


def closure1(unit_var: None, unit_var_1: None) -> str:
    def _arrow127(__unit: None=None, unit_var: Any=unit_var, unit_var_1: Any=unit_var_1) -> bool:
        raise Exception(("platform.is_windows / target: " + str(US1(4, US0(0)))) + "")

    if _arrow127():
        return ".exe"

    else: 
        return ""



def _arrow128(__unit: None=None) -> bool:
    return closure0(None, None)


v0: Callable[[], bool] = _arrow128

def is_windows(__unit: None=None) -> bool:
    return v0(None)


def _arrow129(__unit: None=None) -> str:
    return closure1(None, None)


v1: Callable[[], str] = _arrow129

def get_executable_suffix(__unit: None=None) -> str:
    return v1(None)

File: lib/spiral/platform.ts
import { Union } from "../../deps/Fable/src/fable-library-ts/Types.js";
import { union_type, TypeInfo } from "../../deps/Fable/src/fable-library-ts/Reflection.js";

export type US0_$union = 
    | US0<0>
    | US0<1>
    | US0<2>

export type US0_$cases = {
    0: ["US0_0", []],
    1: ["US0_1", []],
    2: ["US0_2", []]
}

export function US0_US0_0() {
    return new US0<0>(0, []);
}

export function US0_US0_1() {
    return new US0<1>(1, []);
}

export function US0_US0_2() {
    return new US0<2>(2, []);
}

export class US0<Tag extends keyof US0_$cases> extends Union<Tag, US0_$cases[Tag][0]> {
    constructor(readonly tag: Tag, readonly fields: US0_$cases[Tag][1]) {
        super();
    }
    cases() {
        return ["US0_0", "US0_1", "US0_2"];
    }
}

export function US0_$reflection(): TypeInfo {
    return union_type("Platform.US0", [], US0, () => [[], [], []]);
}

export type US1_$union = 
    | US1<0>
    | US1<1>
    | US1<2>
    | US1<3>
    | US1<4>

export type US1_$cases = {
    0: ["US1_0", [US0_$union]],
    1: ["US1_1", [US0_$union]],
    2: ["US1_2", [US0_$union]],
    3: ["US1_3", [US0_$union]],
    4: ["US1_4", [US0_$union]]
}

export function US1_US1_0(f0_0: US0_$union) {
    return new US1<0>(0, [f0_0]);
}

export function US1_US1_1(f1_0: US0_$union) {
    return new US1<1>(1, [f1_0]);
}

export function US1_US1_2(f2_0: US0_$union) {
    return new US1<2>(2, [f2_0]);
}

export function US1_US1_3(f3_0: US0_$union) {
    return new US1<3>(3, [f3_0]);
}

export function US1_US1_4(f4_0: US0_$union) {
    return new US1<4>(4, [f4_0]);
}

export class US1<Tag extends keyof US1_$cases> extends Union<Tag, US1_$cases[Tag][0]> {
    constructor(readonly tag: Tag, readonly fields: US1_$cases[Tag][1]) {
        super();
    }
    cases() {
        return ["US1_0", "US1_1", "US1_2", "US1_3", "US1_4"];
    }
}

export function US1_$reflection(): TypeInfo {
    return union_type("Platform.US1", [], US1, () => [[["f0_0", US0_$reflection()]], [["f1_0", US0_$reflection()]], [["f2_0", US0_$reflection()]], [["f3_0", US0_$reflection()]], [["f4_0", US0_$reflection()]]]);
}

export function closure0(unitVar: void, unitVar_1: void): boolean {
    throw new Error(`platform.is_windows / target: ${US1_US1_3(US0_US0_0())}`);
}

export function closure1(unitVar: void, unitVar_1: void): string {
    if ((() => {
        throw new Error(`platform.is_windows / target: ${US1_US1_3(US0_US0_0())}`);
    })()) {
        return ".exe";
    }
    else {
        return "";
    }
}

export const v0 = (): boolean => closure0(undefined, undefined);

export function is_windows(): boolean {
    return v0();
}

export const v1 = (): string => closure1(undefined, undefined);

export function get_executable_suffix(): string {
    return v1();
}


